### PR TITLE
Drop CentOS 7, current-podified-dev and build-test-packages related code

### PR DESCRIPTION
**Pull request by raukadah**
This pr drops old code from repo-setup
- CentOS 7 is never used in EDPM deployment.
- Build-test-packages role is imported in ci-framework: 
  https://github.com/openstack-k8s-operators/ci-framework/tree/main/ci_framework/roles/build_openstack_packages
- current-podified-dev was added for tripleo land and not used in
  EDPM deployment. More info is here: https://github.com/openstack-k8s-operators/install_yamls/pull/526
- osptools is never used in edpm deployment so dropping it.
- It also removes unsupported old release.
- It also drops INCLUDE_PKGS in the dlrn repos.

Note: CentOS Stream 8 support will be cleaned up in a seperate pr.

### CODE REVIEW

File: plugins/module_utils/repo_setup/get_hash/__main__.py [hunk 1]
```diff
@@ -20,12 +20,6 @@
 import sys
 from repo_setup.utils import load_logging
 from repo_setup.get_hash.hash_info import HashInfo
-import repo_setup.get_hash.exceptions as exc
-
-
-def _validate_args(parsed_args):
-    if parsed_args.os_version == "centos7" and (parsed_args.component is not None):
-        raise exc.HashInvalidParameter("Cannot specify component for centos 7")
```
Comment (viroel) [inline plugins/module_utils/repo_setup/get_hash/__main__.py:28]: If you submit another commit, please remove the comment on line 37:
"This is NOT valid for Centos 7."

**Review by viroel — COMMENTED** [review]
Hi Chandan, you are also removing build-test-packages role from this repo, can you also add a comment about that in PR description or commit message?
Thanks.
Looks good in overall, thanks!

File: plugins/module_utils/repo_setup/get_hash/constants.py [hunk 2]
```diff
@@ -77,9 +67,9 @@
         "security",
         "swift",
         "tempest",
-        "tripleo",
+        "podified",
         "ui",
         "validation",
     ],
-    "os_versions": ["centos7", "centos8", "centos9", "rhel8", "rhel9"],
+    "os_versions": ["centos8", "centos9", "rhel8", "rhel9"],
```
Comment (rlandy) [inline plugins/module_utils/repo_setup/get_hash/constants.py:74]: if you are just keeping podified and osp-18, antelope and master, do we need centos8 and rhel8?

File: tests/unit/repo_setup/test_main.py [hunk 3]
```diff
@@ -611,21 +530,21 @@ def test_inject_mirrors_centos(self):
         start_repo = '''
 [delorean]
 name=delorean
-baseurl=https://trunk.rdoproject.org/centos7/some-repo-hash
+baseurl=https://trunk.rdoproject.org/centos9/some-repo-hash
 enabled=1
 [centos]
 name=centos
-baseurl=http://mirror.centos.org/centos/7/virt/$basearch/kvm-common
+baseurl=http://mirror.centos.org/centos/9/virt/$basearch/kvm-common
```
Comment (rlandy) [inline tests/unit/repo_setup/test_main.py:433]: Doesnt resolve .... https://mirror.stream.centos.org/9-stream/

File: plugins/module_utils/repo_setup/main.py [hunk 4]
```diff
@@ -577,7 +530,6 @@ def install_deps(args, base_path):
             content = _change_priority(content, 20)
             _write_repo(content, args.output_path, name="delorean-current-podified")
             content = _get_repo(base_path + "current/delorean.repo", args)
```
Comment (karelyatin) [inline plugins/module_utils/repo_setup/main.py]: Why current-podified-dev is needed at all now since include pkgs are not used? without it always current repos will be used for all the openstack rpms as that will be higher priority.

**Review by karelyatin — CHANGES_REQUESTED** [review]

File: plugins/module_utils/repo_setup/get_hash/__main__.py (hunk 1 above)
Comment (raukadah) [reply plugins/module_utils/repo_setup/get_hash/__main__.py:28]: We no longer needs this method. It is already removed from line 81.

File: plugins/module_utils/repo_setup/get_hash/constants.py (hunk 2 above)
Comment (raukadah) [reply plugins/module_utils/repo_setup/get_hash/constants.py:74]: We will dropp CentOS/RHEL-8 related stuff in a seperate pr.

File: tests/unit/repo_setup/test_main.py (hunk 3 above)
Comment (raukadah) [reply tests/unit/repo_setup/test_main.py:433]: Test data no longer needed. Dropped from the code.
